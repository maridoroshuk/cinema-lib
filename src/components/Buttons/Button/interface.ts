export interface IButtonProps {
  children: string;
  backgroundColor?: string;
  size?: 'small' | 'medium' | 'large';
  type?: 'submit' | 'button';
  value: string;
  onClick?: () => void;
  dataCy: string;
}
