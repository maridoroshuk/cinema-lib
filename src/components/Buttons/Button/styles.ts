import styled, { css } from 'styled-components';

interface ButtonProps {
  backgroundColor?: string;
  size?: 'small' | 'medium' | 'large';
}

const gold = css`
  background-color: ${({ theme }) => theme.colors.gold};
  color: ${({ theme }) => theme.colors.neutral000};
`;

const white = css`
  background-color: ${({ theme }) => theme.colors.neutral000};
  color: ${({ theme }) => theme.colors.neutral1000};
`;

const small = css`
  border-radius: ${({ theme }) => theme.sizes.borderRadius[10]}px;
`;

const medium = css`
  height: ${({ theme }) => theme.sizes.height[50]}px;
  border-radius: ${({ theme }) => theme.sizes.borderRadius[10]}px;
`;

const large = css`
  font-weight: ${({ theme }) => theme.sizes.fontWeight[700]};
  height: ${({ theme }) => theme.sizes.height[80]}px;
  border-radius: ${({ theme }) => theme.sizes.borderRadius[15]}px;
`;

export const ButtonStyled = styled.button<ButtonProps>`
  width: 100%;
  font-size: clamp(
    ${({ theme }) =>
      `${theme.sizes.fontSize[16]}px, 2.5vw, ${theme.sizes.fontSize[20]}px`}
  );
  border: 0;
  cursor: pointer;
  font-family: ${({ theme }) => theme.fontFamily.Poppins};
  text-align: center;
  padding: ${({ theme }) => theme.spaces[5]}px
    ${({ theme }) => theme.spaces[10]}px;

  ${({ backgroundColor }) => {
    switch (backgroundColor) {
      case 'gold':
        return gold;
      case 'white':
        return white;
      default:
        return gold;
    }
  }}

  ${({ size }) => {
    switch (size) {
      case 'small':
        return small;
      case 'medium':
        return medium;
      case 'large':
        return large;
      default:
        return medium;
    }
  }}


  @media (min-width: ${({ theme }) => theme.screenSizes[600]}px) {
    padding: ${({ theme }) => theme.spaces[10]}px
      ${({ theme }) => theme.spaces[20]}px;
  }
`;
