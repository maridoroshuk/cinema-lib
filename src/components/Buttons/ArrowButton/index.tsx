import React from 'react';

import ArrowImage from '@/assets/icons/arrow.png';

import { Arrow, Wrapper } from './styles';
import { IArrowButton } from './interface';

export const ArrowButton = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Arrow src={ArrowImage} alt="arrow (hide)" />
    </Wrapper>
  );
};
