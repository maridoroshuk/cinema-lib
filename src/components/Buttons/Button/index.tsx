import React from 'react';

import { ButtonStyled } from '@/components/Buttons/Button/styles';

import { IButtonProps } from './interface';

export const Button = ({
  children,
  size,
  backgroundColor,
  type,
  value,
  onClick,
  dataCy,
}: IButtonProps) => {
  return (
    <ButtonStyled
      backgroundColor={backgroundColor}
      size={size}
      type={type || 'button'}
      value={value}
      onClick={onClick}
      data-cy={dataCy}
    >
      {children}
    </ButtonStyled>
  );
};
