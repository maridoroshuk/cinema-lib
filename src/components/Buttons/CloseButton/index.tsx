import React from 'react';
import { ICloseButtonProps } from './interface';
import { ButtonWrapper, StyledButton } from './styles';

export const CloseButton = ({ onClose }: ICloseButtonProps) => {
  return (
    <ButtonWrapper>
      <StyledButton onClick={onClose} />
    </ButtonWrapper>
  );
};
