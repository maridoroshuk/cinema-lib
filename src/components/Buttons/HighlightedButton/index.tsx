import React from 'react';
import { IUnderlineButton } from './interface';
import { ButtonStyled } from './styles';

export const HighlightedButton = ({
  children,
  handleButtonClick,
  value,
}: IUnderlineButton) => {
  return (
    <ButtonStyled onClick={handleButtonClick} value={value || ''}>
      {children}
    </ButtonStyled>
  );
};
