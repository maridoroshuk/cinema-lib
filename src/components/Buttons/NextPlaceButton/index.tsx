import React from 'react';

import NextPlaceButton from '@/assets/icons/next_place_button.png';

import { Next, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const NextPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Next src={NextPlaceButton} />
    </Wrapper>
  );
};
