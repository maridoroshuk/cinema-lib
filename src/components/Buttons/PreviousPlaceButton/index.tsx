import React from 'react';

import PreviousPlaceButton from '@/assets/icons/previous_place_button.png';

import { Previous, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const PreviousPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Previous src={PreviousPlaceButton} />
    </Wrapper>
  );
};
