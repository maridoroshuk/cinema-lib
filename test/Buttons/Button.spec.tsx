import React from 'react';
import { render } from '@testing-library/react';

import { Button } from '@/components/Buttons/Button';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('Button', () => {
  it('renders small button with type button', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <Button size="small" type="button" onClick={handleClick} value="signUp">
          Sign up
        </Button>,
      ),
    );
    expect(container.firstChild).toHaveAttribute('type', 'button');
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders medium button with background white', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <Button
          size="medium"
          backgroundColor="white"
          onClick={handleClick}
          value="signUp"
        >
          Sign up
        </Button>,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders large button with value email and type submit', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <Button size="large" value="email" type="submit" onClick={handleClick}>
          Save
        </Button>,
      ),
    );
    expect(container.firstChild).toHaveAttribute('type', 'submit');
    expect(container.firstChild).toHaveAttribute('value', 'email');
    expect(container.firstChild).toMatchSnapshot();
  });
});
