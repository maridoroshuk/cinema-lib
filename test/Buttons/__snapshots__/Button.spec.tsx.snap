// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders large button with value email and type submit 1`] = `
.c0 {
  border: 0;
  cursor: pointer;
  font-family: 'Poppins',sans-serif;
  text-align: center;
  background-color: #D98639;
  color: #FFFFFF;
  font-size: 24px;
  font-weight: 700;
  padding: 10px 20px;
  width: 100%;
  height: 80px;
  border-radius: 15px;
}

<button
  class="c0"
  type="submit"
  value="email"
>
  Save
</button>
`;

exports[`Button renders medium button with background white 1`] = `
.c0 {
  border: 0;
  cursor: pointer;
  font-family: 'Poppins',sans-serif;
  text-align: center;
  background-color: #FFFFFF;
  color: #000000;
  font-size: 24px;
  padding: 10px 20px;
  width: 100%;
  height: 50px;
  border-radius: 10px;
}

<button
  class="c0"
  type="button"
  value="signUp"
>
  Sign up
</button>
`;

exports[`Button renders small button with type button 1`] = `
.c0 {
  border: 0;
  cursor: pointer;
  font-family: 'Poppins',sans-serif;
  text-align: center;
  background-color: #D98639;
  color: #FFFFFF;
  font-size: 16px;
  padding: 10px 20px;
  width: 100%;
  height: 50px;
  border-radius: 10px;
}

<button
  class="c0"
  type="button"
  value="signUp"
>
  Sign up
</button>
`;
